src/business_logic/fact_state/contract_state.rs

```diff
@@ -1,26 +1,35 @@
+use std::collections::HashMap;
+
 use felt::Felt;
 use serde::{Deserialize, Serialize};
 
-use crate::business_logic::state::cached_state::UNINITIALIZED_CLASS_HASH;
+use crate::{business_logic::state::cached_state::UNINITIALIZED_CLASS_HASH, utils::Address};
 
 #[derive(Deserialize, Serialize, PartialEq, Debug, Clone)]
 pub struct ContractState {
     pub(crate) contract_hash: Vec<u8>,
     pub(crate) nonce: Felt,
+    pub(crate) storage_commitment_tree: HashMap<Felt, Felt>,
 }
 
 impl ContractState {
-    pub(crate) fn create(contract_hash: Vec<u8>, nonce: Felt) -> Self {
+    pub(crate) fn create(
+        contract_hash: Vec<u8>,
+        nonce: Felt,
+        storage_commitment_tree: HashMap<Felt, Felt>,
```

Comment:
we should change the name to something else because we don't have a commitment tree here.

Reply:
changed to `storage_keys`